sql/plan/analyze.go

```diff
@@ -0,0 +1,105 @@
+package plan
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+type Analyze struct {
+	tbls []sql.Node
+}
+
+func NewAnalyze(tbls []sql.Node) *Analyze {
+	return &Analyze{
+		tbls: tbls,
+	}
+}
+
+// Schema implements the interface sql.Node.
+// TODO: should be |Tables|Op|Msg_type|Msg_text|
+func (n *Analyze) Schema() sql.Schema {
+	return sql.OkResultSchema
+}
+
+// String implements the interface sql.Node.
+func (n *Analyze) String() string {
+	tblNames := make([]string, len(n.tbls))
+	for i, tbl := range n.tbls {
+		switch resTbl := tbl.(type) {
+		case *ResolvedTable:
```

Comment:
can other table nodes end up here? do we error earlier if a `QueryProcess` or `DeferredAsofTable` node is here for some reason?